### ISSUE TITLE
Training Loop Task

### ISSUE THREAD

**Issue by abbycakes02** [abbycakes02/cmpm118_chessengine#8]
- Create backend_engine/ml/train.py.
 - Loss Function: Mean Squared Error (MSE) for Value Head.
 - Optimizer: Adam or SGD.
 - Implement checkpoint saving (.pth files) to backend_engine/models/.
